src/app/Http/Controllers/Auth/MyAccountController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Backpack\Base\app\Http\Controllers\Auth;
+
+use Auth;
+use Backpack\Base\app\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Validation\Rule;
+use Validator;
+
+class MyAccountController extends Controller
+{
+    protected $data = [];
+
+    public function __construct()
+    {
+        $this->middleware('admin');
+    }
+
+    public function showAccountInfoForm()
+    {
+        $this->data['title'] = trans('backpack::base.my_account');
+        $this->data['user'] = Auth::user();
+
+        return view('backpack::auth.account.update_info', $this->data);
+    }
+
+    public function showChangePasswordForm()
+    {
+        $this->data['title'] = trans('backpack::base.my_account');
+        $this->data['user'] = Auth::user();
+
+        return view('backpack::auth.account.change_password', $this->data);
+    }
+
+    public function update(Request $request)
+    {
+        $user = Auth::user();
+
+        $validator = Validator::make($request->all(), [
+            'email' => [
+                'required',
+                'email',
+                Rule::unique($user->getTable())->ignore($user->getKey()),
+            ],
+            'name' => 'required',
+        ]);
+
+        if ($validator->fails()) {
+            return redirect()->back()->withErrors($validator)->withInput($request->all());
+        } else {
+            $data = $request->except(['_token']);
+            $user->update($data);
+
+            \Alert::success(trans('backpack::base.account_updated'))->flash();
+
+            return redirect()->back();
+        }
+    }
+
+    public function updatePassword(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
```

Comment:
Should we be enforcing the `required|min:6` as this might not be anywhere else? especially for the old password, we shouldnt enforce logic on that as it prevents cases where users had invalid passwords to update.

Also again maybe we should extract this logic out? I think if we just call smaller helper functions like `$this->getPasswordValidation()` this will allow users to just update the bits they want, rather than overloading the whole method?

Reply:
Totally. I've removed the ```min:6``` and moved the validation to a FormRequest.